microdnf/commands/clean/clean.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "clean.hpp"
+
+#include "microdnf/context.hpp"
+
+#include <libdnf/repo/repo_cache.hpp>
+
+#include <filesystem>
+#include <iostream>
+
+
+namespace fs = std::filesystem;
+
+
+namespace microdnf {
+
+
+namespace {
+
+struct CacheType {
+    std::string name;
+    std::string description;
+    CleanCommand::Actions action;
+};
+
+const CacheType CACHE_TYPES[]{
+    {"all", "Delete all cached data from the repositories cache", CleanCommand::CLEAN_ALL},
+    {"packages", "Delete packages from the repositories cache", CleanCommand::CLEAN_PACKAGES},
+    {"metadata",
+     "Delete the metadata and dbcache from the repositories cache",
+     static_cast<CleanCommand::Actions>(CleanCommand::CLEAN_METADATA | CleanCommand::CLEAN_DBCACHE)},
+    {"dbcache", "Delete dbcache from the repositories cache", CleanCommand::CLEAN_DBCACHE},
+    {"expire-cache", "Mark the repositories cache as expired", CleanCommand::EXPIRE_CACHE}};
+
+}  // namespace
+
+
+using namespace libdnf::cli;
+
+
+CleanCommand::CleanCommand(Command & parent) : Command(parent, "clean") {
+    auto & ctx = static_cast<Context &>(get_session());
+    auto & parser = ctx.get_argument_parser();
+
+    auto & cmd = *get_argument_parser_command();
+    cmd.set_short_description("Remove or expire cached data");
+
+    auto cache_types =
+        parser.add_new_positional_arg("cache_types", ArgumentParser::PositionalArg::AT_LEAST_ONE, nullptr, nullptr);
+    cache_types->set_short_description("List of cache types to clean up");
+
+    cache_types->set_parse_hook_func(
```

Comment:
Please can you add a note what inline does? It tests arguments whether they they are known and set a bit for the action in `required_actions`